packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -80,14 +82,25 @@ function MaybeIframe( {
 }
 
 export default function VisualEditor( { styles } ) {
-	const { deviceType, isTemplateMode } = useSelect( ( select ) => {
+	const {
+		deviceType,
+		isTemplateMode,
+		wrapperBlockName,
+		wrapperUniqueId,
+	} = useSelect( ( select ) => {
 		const {
 			isEditingTemplate,
 			__experimentalGetPreviewDeviceType,
 		} = select( editPostStore );
+		const { getCurrentPostId, getCurrentPostType } = select( editorStore );
 		return {
 			deviceType: __experimentalGetPreviewDeviceType(),
 			isTemplateMode: isEditingTemplate(),
+			wrapperBlockName:
```

Comment:
We would also need to check for `template parts` where the `getCurrentPostType() === 'wp_template_part'`

Reply:
I haven't explored Template Parts yet. It's more complex because they can be loaded from files and then they don't have a corresponding post id and that's why they are handled in a different way in the `edit` implementation:

https://github.com/WordPress/gutenberg/blob/054ff5a7c5319989c32c67c3abdbacd76ebd4c22/packages/block-library/src/template-part/edit/index.js#L36-L40

We will have to refactor Template Part implementation to use a post id when present to pass to `useNoRecursiveRenders` hook.